packages/test/service-load-test/src/loadTestDataStore.ts

```diff
@@ -87,104 +87,146 @@ class LoadTestDataStoreModel {
+        return Date.now() - (this.haveTaskLock() ?  this.taskStartTime : this.startTime);
+    }
+
+    public async getPartnerCounter() {
+        if(this.runtime.disposed) {
+            return undefined;
+        }
+        const dir = this.root.getSubDirectory(this.partnerId.toString());
+        if(dir === undefined) {
+            return undefined;
+        }
+        const handle = dir.get<IFluidHandle<ISharedCounter>>(counterKey);
+        if(handle === undefined) {
+            return undefined;
+        }
+        return handle.get();
     }
 
     public haveTaskLock() {
-        try{
-            return this.taskManager.haveTaskLock(this.taskId);
-        }catch{
-            // remove try catch after taskManager fixes
+        if(this.runtime.disposed) {
             return false;
         }
+        return this.taskManager.haveTaskLock(this.taskId);
     }
 
     public abandonTask() {
         if(this.haveTaskLock()) {
-            try{
-                this.taskManager.abandon(this.taskId);
-            }catch{
-                // remove try catch after taskManager fixes
-            }
+            this.taskManager.abandon(this.taskId);
         }
     }
 
     public async lockTask() {
+        if(this.runtime.disposed) {
+            return;
+        }
         if(!this.haveTaskLock()) {
-            if(!this.runtime.connected) {
-                await new Promise((res,rej)=>{
-                    this.runtime.once("connected",res);
-                    this.runtime.once("dispose", rej);
-                });
-            }
             try{
+                if(!this.runtime.connected) {
+                    await new Promise<void>((res,rej)=>{
+                        const resAndClear = ()=>{
+                            res();
+                            this.runtime.off("connected", resAndClear);
+                            this.runtime.off("disconnected", rejAndClear);
+                            this.runtime.off("dispose", rejAndClear);
+                        };
+                        const rejAndClear = ()=>{
+                            rej(new Error("failed to connect"));
+                            resAndClear();
+                        };
+                        this.runtime.once("connected",resAndClear);
+                        this.runtime.once("dispose",rejAndClear);
+                        this.runtime.once("disconnected",rejAndClear);
+                    });
+                }
                 await this.taskManager.lockTask(this.taskId);
                 this.taskStartTime = Date.now();
-                this.taskManager.once("lost",(taskId)=>{
-                    if(taskId === this.taskId) {
-                        this.dir.set(taskTimeKey, Date.now() - this.taskStartTime);
-                        this.taskStartTime = 0;
-                    }
-                });
-            }catch{
-                // remove try catch after taskManager fixes
+            }catch(e) {
```

Comment:
What error do you expect here when the runtime is disposed or disconnected? Is it worth logging even?

Reply:
runner.ts will log it and push it to the logger. i prefer to only handle unknown error at the top of the stack. keeps from logging duplicates, and ensures fast fail